### PR TITLE
Reflection_Engine: InheritedTypes stopped from passing nulls to output

### LINKED ISSUE #2330: Shouldn't InheritedTypes exclude nulls?
#### Description:
While working on #2327 I have noticed that the lines below may lead to null in the returned set:
https://github.com/BHoM/BHoM_Engine/blob/2761083ed5771eee3c642544cbc73e9f1e5f8b6f/Reflection_Engine/Query/InheritedTypes.cs#L42-L43

Wouldn't it be good to add a null check on `type.BaseType` before adding it to `types`?

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #2330

### Test files

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FBHoM%5FEngine%2FReflection%5FEngine%2F%232402%2DNullInInheritedTypes%2Egh&parent=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FBHoM%5FEngine%2FReflection%5FEngine).

### Changelog

- InheritedTypes stopped from passing nulls to output

### CODE REVIEW

File: Reflection_Engine/Query/InheritedTypes.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public static List<Type> InheritedTypes(this Type type, bool onlyBHoM = false)
         {
             List<Type> types = type.GetInterfaces().ToList();
 
-            if (type.BaseType != typeof(object))
+            if (type.BaseType != null && type != typeof(object))
```
Comment (adecler) [inline Reflection_Engine/Query/InheritedTypes.cs:42]: Shouldn't it be `(type.BaseType != null && type.BaseType != typeof(object))` ?

**Review by adecler — CHANGES_REQUESTED** [review]

File: Reflection_Engine/Query/InheritedTypes.cs (hunk 1 above)
Comment (pawelbaran) [reply Reflection_Engine/Query/InheritedTypes.cs]: Of course it should be 🤦 I have just committed a fix, thanks for being careful!

**Review by adecler — APPROVED** [review]
Better now 😉

**Comment by pawelbaran** [comment]
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check core

**Comment by pawelbaran** [comment]
@BHoMBot check installer

**Comment by pawelbaran** [comment]
@BHoMBot check versioning

**Comment by pawelbaran** [comment]
@BHoMBot check serialisation

**Comment by pawelbaran** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**